addons/addon-base-raas-ui/packages/base-raas-ui/src/models/aws-accounts/AwsAccount.js

```diff
@@ -53,6 +54,13 @@ const AwsAccount = types
       self.updatedAt = rawAwsAccounts.updatedAt || self.updatedAt;
       self.createdBy = rawAwsAccounts.createdBy || self.createdBy;
       self.updatedBy = rawAwsAccounts.updatedBy || self.updatedBy;
+      self.needsPermissionUpdate =
```

Comment:
Yes, I was envisioning the flow to be:
- user is added to SWB, `needsPermissionUpdate === undefined` since it hasn't been onboarded yet
- user is onboarded, `needsPermissionUpdate` is set to either `true` or `false`
- once onboarded, `needsPermissionUpdate` should only ever be a boolean

Reply:
I wasn't sure if there was a better way to implement this...
The other maybe values used a double check for permissions and I was emulating those, but it may be simpler to skip the second check and just default to `self.needsPermissionUpdate`, since if it's missing it'll be equal to `undefined` anyway. I'll make that change.

Reply:
Yes, the flow makes sense. BYOB account go through the same flow as well, might be worth checking how it's handled there. Normally we want to avoid using `undefined` or `null` for an actual meaningful status, since it's confusing for readers and it could be the result of an unintended event(exception or something else).

Reply:
gotcha. Maybe it would be better to just use a string arg with predefined entries

Reply:
I looked at BYOB and made edits based on how theirs is implemented--`needsPermissionUpdate` is now strictly a boolean, and checking for onboarding will look at the `cfnStackName` parameter, which will be created and added when the account is onboarded.